src/nvim/api/mark_extended.c

```diff
@@ -0,0 +1,163 @@
+/*
+ * Same function but names are not limited to one char
+ * There is also no ex_command, just viml functions
+ */
+
+#include "nvim/vim.h"
+#include "nvim/globals.h"      // FOR_ALL_BUFFERS
+#include "nvim/mark.h"         // SET_FMARK
+#include "nvim/memory.h"
+#include "nvim/map.h"          // pmap ...
+#include "nvim/lib/kbtree.h"   // kbitr ...
+#include "nvim/api/mark_extended.h"
+
+#ifdef INCLUDE_GENERATED_DECLARATIONS
+# include "api/extmark.c.generated.h"
+#endif
+
+#define FOR_ALL_EXTMARKS(buf) \
+  kbitr_t itr; \
+  ExtendedMark *extmark; \
+  for (;kb_itr_valid(&itr); kb_itr_next(ExtendedMark, buf->b_extmarks_tree, &itr)){
+
+/* Create or update an extmark, */
+int extmark_set(buf_T *buf, cstr_t *name, pos_T *pos)
```

Comment:
This is rather strange to see `cstr_t` anywhere, but where it is absolutely required because `char *` cannot be part of the function/etc name. It should be `char **` if you really meant `cstr_t *` (`cstr_t` is an alias to `char *`, not `char`), just `char *` otherwise.